test/org/javarosa/model/xform/SmsSerializingVisitorTest.java

```diff
@@ -10,123 +10,110 @@
 
 import java.io.File;
 import java.io.IOException;
-import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
-import static org.javarosa.test.utils.Utils.convertToString;
 import static org.junit.Assert.assertFalse;
-import static org.junit.Assert.assertTrue;
 import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
 
 /**
  * Performs a comprehensive suite of tests on the results of a form
  * being serialized to the SMS format.
  */
-public class SMSSerializingVisitorTest {
-    private FormParseInit formParser;
-    private String SMS;
+public class SmsSerializingVisitorTest {
+    private String sms;
     private static final String delimiter = " ";
-    private FormEntryController formEntryController;
-    private FormInstance formInstance;
 
     @Before
     public void setup() throws IOException {
-        formParser = new FormParseInit();
+        FormParseInit formParser = new FormParseInit();
         formParser.setFormToParse(new File(PathConst.getTestResourcePath(), "sms_form.xml").toString());
-        formEntryController = formParser.getFormEntryController();
-        formInstance = formEntryController.getModel().getForm().getInstance();
+        FormEntryController formEntryController = formParser.getFormEntryController();
+        FormInstance formInstance = formEntryController.getModel().getForm().getInstance();
 
         SMSSerializingVisitor serializer = new SMSSerializingVisitor();
 
         ByteArrayPayload payload = (ByteArrayPayload) serializer.createSerializedPayload(formInstance);
 
-        SMS = convertToString(payload.getPayloadStream());
-        SMS = SMS.replace("\\", "").replace("\\\\", "\\");
+        sms = new String(payload.getPayloadBytes(), "UTF-8").replace("\\", "").replace("\\\\", "\\");
     }
 
     @Test
     public void SmsNotNull() {
-        assertNotNull(SMS);
+        assertNotNull(sms);
     }
 
     @Test
     public void testTagsAreBeingSerialized() {
         for (String tag : getAnswerTags()) {
-            assertTrue(SMS.contains(tag));
+            assertTrue(sms.contains(tag));
         }
     }
 
     /**
-     * Checks to see if all answers for the selected tags exist. This
+     * Checks to see if all answers for the selected tags exist.This
      * targets answers of various data types  and structures such as
      * groups.
      */
     @Test
     public void testTaggedAnswers() {
         for (String tag : getAnswerTags()) {
-            assertTrue(taggedAnswerExists(tag, SMS));
+            assertTrue(taggedAnswerExists(tag, sms));
         }
     }
 
     /***
-     * Checks to see an answer that doesn't have a tag is present .
-     * Placeholder  is an answer bounded to the <maiden_name> tag
+     * Checks to see than an answer doesn't have a tag is present.
+     * Placeholder is an answer bounded to the <maiden_name> tag
```

Comment:
“bounded” has a different meaning than “bound” (past tense of “bind”), which is what you probably mean.